native/packages/react-native-bpk-component-chip/src/BpkChip.js

```diff
@@ -0,0 +1,147 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* @flow */
+
+import React from 'react';
+import {
+  Platform,
+  View,
+  ViewPropTypes,
+  StyleSheet,
+  type StyleObj,
+} from 'react-native';
+import PropTypes from 'prop-types';
+import BpkText from 'react-native-bpk-component-text';
+import BpkTouchableNativeFeedback from 'react-native-bpk-component-touchable-native-feedback';
+import BpkTouchableOverlay from 'react-native-bpk-component-touchable-overlay';
+import {
+  borderRadiusSm,
+  colorBlue700,
+  colorGray100,
+  colorGray700,
+  colorWhite,
+  spacingSm,
+  spacingMd,
+} from 'bpk-tokens/tokens/base.react.native';
+
+const styles = StyleSheet.create({
+  wrapper: {
+    backgroundColor: colorGray100,
+    borderRadius: borderRadiusSm,
+    padding: spacingSm,
+  },
+  inner: {
+    alignItems: 'center',
+    flexDirection: 'row',
+    justifyContent: 'space-between',
+  },
+  innerLarge: {
+    paddingVertical: spacingMd,
+  },
+  wrapperSelected: {
+    backgroundColor: colorBlue700,
+  },
+  text: {
+    color: colorGray700,
+  },
+  textSelected: {
+    color: colorWhite,
+  },
+});
+
+type Props = {
+  accessibilityLabel: string,
+  onPress: () => mixed,
+  label: string,
+  large: boolean,
+  selected: boolean,
+  style: ?StyleObj,
+};
+
+const BpkChip = (props: Props) => {
+  const {
+    accessibilityLabel,
+    label,
+    large,
+    onPress,
+    selected,
+    style,
+    ...rest
+  } = props;
+
+  const wrapperStyle = [styles.wrapper];
+  const innerStyle = [styles.inner];
+  const textStyle = [styles.text];
+
+  if (large) {
+    innerStyle.push(styles.innerLarge);
+  }
+
+  if (selected) {
+    wrapperStyle.push(styles.wrapperSelected);
+    textStyle.push(styles.textSelected);
+  }
+
+  if (style) {
+    wrapperStyle.push(style);
+  }
+
+  const touchableProps: { [string]: any } = {
+    accessibilityLabel,
+  };
+  let Touchable;
```

Comment:
Can we use `Platform.select` and a `const` value here instead of this method?

Reply:
Yeah there probably is a tiny one, but thinking about that is premature optimization. You can write it with a single `Platform.select` as

```javascript
const { Touchable, touchableProps } = Platform.select({
  android: {
    Touchable: BpkTouchableNativeFeedback,
    touchableProps: {
      accessibilityLabel,
      accessibilityComponentType: "button",
      color: "red"
    }
  },
  ios: {
    Touchable: BpkTouchableOverlay,
    touchableProps: {
      accessibilityLabel,
      accessibilityTraits: selected ? ["button", "selected"] : ["button"],
      borderRadius: "sm"
    }
  }
});

```